app/src/main/java/com/kinkong/QuestionActivity.java

```diff
@@ -67,11 +74,25 @@ protected void onStop() {
         releaseMediaPlayer();
     }
 
-    public void closeQuestion(View view) {
-        Intent countDownIntent = CountDownActivity.getIntent(this);
-        if (startScreen(countDownIntent)) {
-            finish();
-        }
+    public void closeScreen(long waitTime) {
+        shouldMoveToScreen = true;
+        root.postDelayed(() -> {
+            if (hasNextQuestion) {
+                startNextQuestionVideo();
+            } else {
+                startNextSessionCountDown();
+            }
```

Comment:
no real need for the 2 private methods (startNextQuestionVideo & startNextSessionCountDown). You could simply call startScreen with the correct intent directly from the if/else clause.
finish() can be called outside the if/else clause since it needs to be called for both cases